doctests with ghc-9.12.2 can't parse

**Describe the bug**
We run quickjobs with one version of GHC:

https://github.com/haskell/cabal/blob/fb32f3d1c06d1731b474908e4b86559b202a3d7e/.github/workflows/quick-jobs.yml#L18

I tested doctests with later versions and running doctests on `Cabal` fails with:

```
Can't parse reexported module flag: Language.Haskell.Extension\ as\ Language.Haskell.Extension
```

**To Reproduce**

```
$ make doctest-install
cabal install doctest --overwrite-policy=always --ignore-project --flag cabal-doctest
...

$ doctest --version
doctest version 0.24.3
using version 9.12.2 of the GHC API

$ cd Cabal

$ cabal doctest --allow-newer=False
doctest version 0.24.3
using version 9.12.2 of the GHC API
...
Preprocessing library for Cabal-3.17.0.0...
doctest-0.24.3: Uncaught exception ghc-9.12.2-9b33:GHC.Utils.Panic.GhcException:

Can't parse reexported module flag: Language.Haskell.Extension\ as\ Language.Haskell.Extension

While handling Can't parse reexported module flag: Language.Haskell.Extension\ as\ Language.Haskell.Extension

HasCallStack backtrace:
  throwIO, called at src/Run.hs:152:12 in doctest-0.24.3-bfacc00613da87380f139bf815745504e2634b62b9154c5a8f287d9e3e0f1020:Run
```

**Expected behavior**
That we could run `make doctest` with `ghc-9.12.2`.

**System information**
Ubuntu 25.10, ghc-9.12.2, cabal-3.16.1.0

I see `doctest-0.25.0` was released 2026-05-03 with "Full GHC 9.14 compatibility / -unit-support" from its changelog.

Trying this version with our makefile targets passes for `ghc-9.12.4` but fails for `ghc-9.14.1`:

```
$ git rev-parse HEAD
e6ff0ccf6f2761f1d3d82b4383b053c21f816e27

$ cabal update --ignore-project

$ make doctest-install
...
Build profile: -w ghc-9.14.1 -O1
In order, the following will be built (use -v for more details):
 - doctest-0.25.0 (lib) (requires build)
 - doctest-0.25.0 (exe:doctest) (requires build)
 - doctest-0.25.0 (exe:cabal-doctest) (requires build)
...


$ make doctest
...
Preprocessing library for Cabal-3.17.0.0...
/home/.../cabal/Cabal/src/Distribution/Verbosity.hs:452: \
  failure in expression `import Test.QuickCheck (Arbitrary (..), arbitraryBoundedEnum)'
expected: 
 but got: <no location info>: error: [GHC-45102]
          ^
              Ambiguous module name ‘Test.QuickCheck’.
              it was found in multiple packages:
              QuickCheck-2.18.0.0 QuickCheck-2.18.0.0
          

Examples: 25  Tried: 17  Errors: 0  Failures: 1
make: *** [Makefile:156: doctest] Error 1
```

--- a/.github/workflows/quick-jobs.yml
+++ b/.github/workflows/quick-jobs.yml
@@ -15,7 +15,7 @@ on:
       - created
 
 env:
-  GHC_FOR_QUICK_JOBS: 9.10.3
+  GHC_FOR_QUICK_JOBS: 9.12.2
 
 jobs:
   meta:

--- a/Makefile
+++ b/Makefile
@@ -151,15 +151,21 @@ ghcid-cli: ## Run ghcid for the cabal-install executable.
 
 .PHONY: doctest
 doctest: ## Run doctests.
+	cabal --numeric-version
 	cd Cabal-syntax && $(DOCTEST)
 	cd Cabal-described && $(DOCTEST)
 	cd Cabal && $(DOCTEST)
 	cd cabal-install-solver && $(DOCTEST)
 	cd cabal-install && $(DOCTEST)
 
+# If we pin and periodically bump the version of doctest, we get more
+# reproducible testing. Initially we pinned to doctest-0.25.0 to avoid failures
+# with doctest-0.24.3 but `cabal install doctest` depends on index state that
+# itself gets bumped when `cabal update --ignore-project` is run.
+# SEE: https://github.com/haskell/cabal/issues/11493#issuecomment-4615438425
 .PHONY: doctest-install
 doctest-install: ## Install doctest tool needed for running doctests.
-	cabal install doctest --overwrite-policy=always --ignore-project --flag cabal-doctest
+	cabal install doctest-0.25.0 --overwrite-policy=always --ignore-project --flag cabal-doctest
 
 # tests
 

